frontend/src/metabase/public/containers/PublicAction/PublicAction.tsx

```diff
@@ -33,13 +37,17 @@ function PublicAction({ action, publicId, onError }: Props) {
   const handleSubmit = useCallback(
     async (values: ParametersForActionExecution) => {
       try {
-        await PublicApi.executeAction({ uuid: publicId, parameters: values });
+        const fieldSettings =
+          action.visualization_settings?.fields ||
+          generateFieldSettingsFromParameters(action.parameters);
+        const parameters = setNumericValues(values, fieldSettings);
```

Comment:
Can you elaborate on this bit more? I think this has something to do with the [very recent discussion we have on Slack](https://metaboat.slack.com/archives/C04FG88HL95/p1676488798841259) that you initially thought implicit actions might be impossible to execute publicly or something like that.